### PR TITLE
oci: cleanup log path if the container failed to create

### PR DESCRIPTION
**Pull request by haircommander**
Signed-off-by: Peter Hunt <pehunt@redhat.com>

#### What type of PR is this?

/kind bug
#### Does this PR introduce a user-facing change?

```release-note
fix a bug where CRI-O would leak a log file if a container failed to be created and the pod hadn't yet been cleaned up. 
```

### CODE REVIEW

**Comment by mrunalp** [comment]
integration failure:
```
not ok 53 ctr log cleaned up if container create failed
# (in test file ./ctr.bats, line 184)
#   `ls "$DEFAULT_LOG_PATH/$pod_id"/*' failed with status 2


```

**Comment by haircommander** [comment]
> integration failure:
> 
> ```
> not ok 53 ctr log cleaned up if container create failed
> # (in test file ./ctr.bats, line 184)
> #   `ls "$DEFAULT_LOG_PATH/$pod_id"/*' failed with status 2
> ```

oopsies, inverted boolean expression

**Comment by rphillips** [comment]
lgtm

**Review by mrunalp — APPROVED** [review]
/lgtm

**Comment by littlejawa** [comment]
@haircommander - the kata-jenkins job keeps failing on the test you added.
I could verify it locally, and there is indeed a couple log files remaining (one which name matches the pod_id, and another one with a different ID).

The cleanup code you added in runtime_vm.go should have removed the first at least right?
I'm not sure about the other though.

**Comment by littlejawa** [comment]
How can we prevent Github from repeating a CI job again and again?
The CI is actually reporting an issue. Until we fix it, it will just keep failing.

**Comment by haircommander** [comment]
/hold

Will do the trick

File: internal/oci/runtime_vm.go
Comment (littlejawa) [inline internal/oci/runtime_vm.go:754]: Should this be in CreateContainer() rather than createContainerIO() ?

Comment (haircommander) [reply internal/oci/runtime_vm.go]: hahaha right! fixed :sweat_smile:

File: internal/oci/runtime_oci.go
Comment (TomSweeneyRedHat) [inline internal/oci/runtime_oci.go:205]: nit
```suggestion
				log.Warnf(ctx, "Failed to remove log path %s after failure to create container: %v", c.logPath, err)
```
or perhaps "failing"

File: internal/oci/runtime_vm.go
Comment (TomSweeneyRedHat) [inline internal/oci/runtime_vm.go:138]: if you change prior, change there too.

**Comment by haircommander** [comment]
/hold cancel

rebased to fix ci PTAL @cri-o/cri-o-maintainers

**Pull request merged**
